### PR TITLE
More efficient frame copying.

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
Just saw your tweet asking me for assistance with the ImageSharp API. This is my first time writing F# so I'm hoping I got it right!

I replaced the `[x, y]` frame iteration with a more efficient cloning. I also fixed the leak for the initial gif.

I did spot something else but I'm not confident in my F# to do a fix.

I wouldn't return a byte array from the `getFrameFromGif` method, I'd return the stream and copy it to a filestream in order to remove the overhead of allocating and copying a new array for each time.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ Copyright (C) 2021 FreezeFrame
 
 ## Licenses For ImageSharp
 
-- This tool uses [ImageSharp](https://sixlabors.com/products/imagesharp/), which requires a license for commercial use.
+- This tool uses [ImageSharp](https://sixlabors.com/products/imagesharp/), which requires a license for commercial support.
```
Comment (JimBobSquarePants) [inline readme.md:31]: People can use the libraries commercially under the Apache license. They just can't ask me for any support unless they purchase a Six Labors license.

File: readme.md [hunk 2]
```diff
@@ -39,4 +39,4 @@ Permission is hereby granted, free of charge, to any person obtaining a copy of
 
 The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 
-THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
\ No newline at end of file
+THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```
Comment (JimBobSquarePants) [inline readme.md:42]: Must be a line endings thing

File: Program.fs [hunk 3]
```diff
@@ -10,13 +10,10 @@ type options = {
...
-    let gif = Image.Load(path)
-    let frame = gif.Frames.RootFrame
-    use image = new Image<Rgba32>(frame.Width, frame.Height)
-    let source = frame :?> ImageFrame<Rgba32>
-    for y in 0..frame.Height-1 do
-        for x in 0..frame.Width-1 do
-            image.[x,y] <- source.[x,y]
+    use gif = Image.Load(path)
+    use image = new Image<Rgba32>(gif.Width, gif.Height)
+    image.Frames.AddFrame(gif.Frames.RootFrame)|> ignore;
```
Comment (isaacabraham) [inline Program.fs]: `;` is optional in F# in this case.

**Comment by baronfel** [comment]
@JimBobSquarePants left an MR [here](https://github.com/JimBobSquarePants/FreezeFrame/pull/1) showing how stream copying would be done, per your comment.

File: Program.fs (hunk 3 above)
Comment (JimBobSquarePants) [reply Program.fs]: Took me two reads to figure out what you meant there! 🤣

**Comment by khalidabuhakmeh** [comment]
This is awesome. Thanks so much!

File: Program.fs (hunk 3 above)
Comment (isaacabraham) [reply Program.fs]: Oops :-)

**Pull request merged**